CMakeLists.txt

```diff
@@ -1,21 +1,108 @@
 cmake_minimum_required(VERSION 3.16.0 FATAL_ERROR) # Configurable policies: <= CMP0097
 
-cmake_policy(SET CMP0091 NEW)
-cmake_policy(SET CMP0092 NEW)
-
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
-include(LibtorrentMacros)
-
-read_version("${CMAKE_CURRENT_SOURCE_DIR}/include/libtorrent/version.hpp" VER_MAJOR VER_MINOR VER_TINY)
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
 
 project(libtorrent
-	DESCRIPTION "Bittorrent library"
-	VERSION ${VER_MAJOR}.${VER_MINOR}.${VER_TINY}
+	VERSION "1.2.14"
```

Comment:
Why have you removed reading from version header file?

Reply:
@elgatito

I did not think there was a good reason for CMake to do it that way, since it is trivial to bump the version here every time a "bump version to X.Y.Z" commit is made.

Plus the other build tools also do it this way: https://github.com/arvidn/libtorrent/blob/RC_1_2/build_dist.sh, `https://github.com/arvidn/libtorrent/blob/b027abb6f5ab5625261c7aef99bec3c542f0785c/Jamfile#L15`.

No need for the extra complexity of parsing it from somewhere, which just bloats the CMake scripts.

Reply:
would you mind adding this case to `tools/set_version.py`, to update this version number?